extensions/resource-deployment/src/ui/deployAzureSQLVMWizard/deployAzureSQLVMWizard.ts

```diff
@@ -0,0 +1,199 @@
+		} catch (error) {
+			vscode.window.showErrorMessage(error);
+		}
+	}
+
+	private setEnvironmentVariables(env: NodeJS.ProcessEnv): void {
+		env['AZDATA_NB_VAR_AZURE_SQLVM_PASSWORD'] = this.model.vmPassword;
+		env['AZDATA_NB_VAR_AZURE_SQLVM_SQL_PASSWORD'] = this.model.sqlAuthenticationPassword;
+	}
+
+	public async getRequest(url: string): Promise<any> {
+		let token = this.model.securityToken.token;
+		const config: AxiosRequestConfig = {
+			headers: {
+				'Content-Type': 'application/json',
+				'Authorization': `Bearer ${token}`
+			},
+			validateStatus: () => true // Never throw
+		};
+		const response = await axios.get(url, config);
+		return response;
+	}
+
+	public createFormRowComponent(view: azdata.ModelView, title: string, description: string, component: azdata.Component, required: boolean): azdata.FlexContainer {
+
+		component.updateProperties({
+			required: required,
+			width: '480px'
+		});
+
+		const labelText = view.modelBuilder.text()
+			.withProperties<azdata.TextComponentProperties>(
+				{
+					value: title,
+					width: '250px',
+					description: description,
+					requiredIndicator: required,
+				})
+			.component();
+
+		labelText.updateCssStyles({
+			'font-weight': '400',
+			'font-size': '13px',
+		});
+
+		const flexContainer = view.modelBuilder.flexContainer()
+			.withLayout(
+				{
+					flexFlow: 'row',
+					alignItems: 'center',
+				})
+			.withItems(
+				[labelText, component],
+				{
+					CSSStyles: { 'margin-right': '5px' }
+				})
+			.component();
+		return flexContainer;
+	}
+
+	public changeComponentDisplay(component: azdata.Component, display: ('none' | 'block')) {
+		component.updateProperties({
+			required: display === 'block'
+		});
+		component.updateCssStyles({
+			display: display
+		});
+	}
+
+	public changeRowDisplay(container: azdata.FlexContainer, display: ('none' | 'block')) {
+		container.items.map((component) => {
+			component.updateProperties({
+				required: (display === 'block'),
+			});
+			component.updateCssStyles({
+				display: display,
+			});
+		});
+	}
+
+	public addDropdownValues(component: azdata.DropDownComponent, values: azdata.CategoryValue[], width?: number) {
+		component.updateProperties({
+			values: values,
+			width: '480px'
+		});
+	}
+
+	public showErrorMessage(message: string) {
+		this.wizardObject.message = {
+			text: message,
+			level: azdata.window.MessageLevel.Error
+		};
+	}
+
+	public validatePassword(password: string): string {
+		/**
+		 * 1. Password length should be between 12 and 123.
+		 * 2. Password must have 3 of the following: 1 lower case character, 1 upper case character, 1 number, and 1 special character.
+		 */
+
+		let errorMessage = '';
+
+		if (password.length < 12 || password.length > 123) {
+			errorMessage += 'Password must be between 12 and 123 characters long.\n';
+		}
+
+		let charTypeCounter = 0;
+
+		if (new RegExp('.*[a-z].*').test(password)) {
+			charTypeCounter++;
```

Comment:
append the specific error in each of these conditionals

Reply:
Here, the condition is 'Password must have 3 of the following: 1 lower case character, 1 upper case character, 1 number, and 1 special character.' 

It won't make sense to add error in each case as we need only 3 out of 4.

Reply:
What I mean is if it doesn't meet the requirement in that conditional, then just append it to the error about whats missing in the password.